Feature/BJS2-40530 - event organization

ZhubanyshZh opened the pull request:
added functionals:
-create event
-get event by event id
-delete event by event id
-get events by filter
-update event
-get owned events by user id
-get participated events by user id

ZhubanyshZh committed:
feat(BJS2-40530): event organization

added crud operations with Event

ZhubanyshZh committed:
feat(BJS2-40530): event organization test

added tests for Event service with mockito test and fix some moments

ZhubanyshZh committed:
fix(BJS2-40530): event organization test

fix some moments

ZhubanyshZh committed:
fix(BJS2-40530): event organization test

fix some moments

ZhubanyshZh committed:
fix(BJS2-40530): event organization test

fix some moments in controller

frs717 commented on src/main/java/school/faang/user_service/config/event/EventFilterConfig.java:
```diff
@@ -0,0 +1,27 @@
+        List<EventFilter> filters = new ArrayList<>();
+        filters.add(eventTitleFilter);
+        filters.add(eventUserIdFilter);
+        eventService.setEventFilters(filters);
```
Можно самостоятельно не set'ать список в бин сервиса. Спринг сам подставит его
`@Bean(value = "eventFilters")`

frs717 commented on src/main/java/school/faang/user_service/config/event/EventFilterConfig.java:
```diff
@@ -0,0 +1,27 @@
+    private final EventService eventService;
+
+    @Bean(value = "eventFilters")
+    public List<EventFilter> getEventFilters(EventTitleFilter eventTitleFilter, EventUserIdFilter eventUserIdFilter) {
```
Достаточно в сервисе объявить` List<EventFilter>`. И спринг сам подтянет все бины типа `EventFilter `в этот список

ZhubanyshZh committed:
fix(BJS2-40530): event organization test

-fix config of bean by name eventFilters, deleted the setter
-fix the controller method validation exception handler more correctly, created the class ControllerMethodArgumentValidationExceptionHandle

ZhubanyshZh committed:
fix(BJS2-40530): event organization test

-fix bug because, int no way to null, so change the int to Integer with default value 1

ZhubanyshZh committed:
fix(BJS2-40530): event organization test

-change some moments

ZhubanyshZh committed:
fix(BJS2-40530): event organization filter

-change some moments in filters

MiraGlaceon commented on src/main/java/school/faang/user_service/controller/event/EventController.java:
```diff
@@ -0,0 +1,58 @@
+public class EventController {
+    private final EventService eventService;
+
+    public EventDto create(@Valid EventDto eventDto) {
```
Ты, Жубаныш, достаточно опытный в спринге. Давай попробуем сделать все по красоте и добавить ResponceEntity и мапперы к эндпоинтам.

ZhubanyshZh replied:
я кажется не понял что мапперы к эндпойнтам, Ты подразумеваешь что в параметры методов сервиса класть не сам энтити а дто или вешать эндпойнтам аннотаций как @GetMapping, @DeleteMapping? скорее всего аннотацию нужно вешать да?

ZhubanyshZh replied:
@MiraGlaceon  можешь после ревью назначить другую задачу

MiraGlaceon replied:
Да, я имел ввиду аннотации GetMapping, PostMapping и тд

ZhubanyshZh committed:
fix(BJS2-40530): adding ResponseEntity, Handlers

-made controller return ResponseEntity
-added Handlers for effective hand the exception and return understandable response to front

ZhubanyshZh committed:
fix(BJS2-40530): request mappers

added mappers to event controller endpoints

ZhubanyshZh committed:
fix(BJS2-40530):RequestBody,Ex Handler,Test

change validation handler because returns in response include stack trace

ZhubanyshZh committed:
Merge branch 'Medusa-master-stream7' into feature/BJS2-40530-event-organization

# Conflicts:
#	src/main/java/school/faang/user_service/exception/DataValidationException.java
